middlewares/validators/checkEmail.js

```diff
@@ -0,0 +1,24 @@
+const User = require('../../lib/modelManagers/usermodel');
+
+/**
+ * @description Check if email is already in the database
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next middleware in the stack
+ * @returns {object} res object
+ */
+async function checkEmail(req, res, next) {
+  try {
+    const record = await User.findEmail(req.body.email);
+    if (record.dataValues.email) {
+      return res.status(404).json({
+        status: false,
+        message: 'Authentication Failed',
```

Comment:
`'Authentication Failed'` is not a descriptive error message. A proper error message should be able to tell the user what caused the failure and ways of fixing the error if an action is required from the user to fix the error.

Reply:
fixed